components/track/TrackDetails.vue

```diff
@@ -1,20 +1,71 @@
 <script setup lang="ts">
 import { TrackModel } from "@bcc-code/bmm-sdk-fetch";
 
+const { t } = useI18n();
+
 defineProps<{
   track: TrackModel;
 }>();
 </script>
 
 <template>
-  <div class="py-4 text-lg">
-    {{ track.title }}
+  <div class="relative flex items-center justify-normal gap-3">
+    <div v-if="track.meta?.attachedPicture" class="relative w-24">
+      <ProtectedImage
+        :src="track.meta?.attachedPicture"
+        alt=""
+        class="aspect-square w-24 rounded-md bg-background-2"
+      />
+    </div>
+    <div class="py-4 text-lg">
+      {{ track.title }}
+      <div class="text-on-color-2">
+        {{ track.meta?.artist }}
+      </div>
+    </div>
+  </div>
+  <br />
+  <hr class="bg-background-2 border-0 h-[2px]" />
+  <div class="py-4 text-lg flex">
+    <div class="w-40">
+      <b>{{ t("track.details.album") }}</b>
+    </div>
+    <div>{{ track.meta?.album }}</div>
+  </div>
+  <hr class="bg-background-2 border-0 h-[2px]" />
+  <div class="py-4 text-lg flex">
+    <div class="w-40">
+      <b>{{ t("track.details.publish-date") }}</b>
+    </div>
+    <div>
+      {{ track.publishedAt?.toLocaleDateString() }}
```

Comment:
Thanks for using a localized date representation 🎉 It's just debatable whether a user wants to see it this way when choosing another language - but that's again language vs locale - which is worth a separate discussion.